### PR TITLE
ci: run gke commands via a pod and not from binary

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

**Comment by nebril** [comment]
needs https://github.com/cilium/cilium-cli/pull/181

**Comment by nebril** [comment]
GKE build passed! https://github.com/cilium/cilium-cli/pull/180/checks?check_run_id=2419590535

I will mark it as ready for review, let's keep in mind that we want to move this workflow from `pull_request` to `pull_request_target` before merging. Will do that after reviews are in.

File: .github/workflows/gke.yaml [hunk 1]
```diff
@@ -20,19 +20,6 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
```
Comment (aanm) [inline .github/workflows/gke.yaml:21]: I think we can also remove this line so it is explicit that we are not doing anything from any line of code?

**Review by aanm — CHANGES_REQUESTED** [review]
LGTM, only one minor request

File: .github/workflows/gke.yaml (hunk 1 above)
Comment (nebril) [reply .github/workflows/gke.yaml:21]: We need to check out to get access to master version of the `.github/get-kubeconfig.sh` and helm chart.

Comment (aanm) [reply .github/workflows/gke.yaml:21]: oh I miss that, you are right.

**Review by aanm — APPROVED** [review]

**Pull request merged**